Automattic/wp-calypso
# Media: Manage media scale through Redux preferences state

Related: #8270

This pull request seeks to migrate the media scale preference from the legacy Flux implementation to the newer Redux-based preferences state. In doing so, it introduces a new standalone component for rendering the scale control, which is an app component controlling the preference value. This value is injected directly into the `<MediaLibraryList />` component where it is used.

`<MediaLibrary />` is one of two components still using `<PreferencesData />`, the other being `<PollInvitation />` slated to be removed in #8270. If #8270 is merged before this, I will rebase and remove the `<PreferencesData />` component.

__Testing Instructions:__

Verify that there are no regressions in behavior of the [post editor](http://calypso.localhost:3000/post) media library scale control at both desktop and mobile viewport sizes. Notably, you should confirm that the media scale allows smooth scrolling between predefined scale options and that a network request to save the preference is triggered after a one second delay. Upon refreshing, your preference should be persisted.

Ensure that Mocha tests pass:

```
npm run test-client
```

cc @deBhal @nylen
